ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -121,7 +130,126 @@ public static ASTNode convert(final RelNode relNode, List<FieldSchema> resultSch
     return c.convert();
   }
 
+  /**
+   * This method generates the abstract syntax tree of a query does not return any rows.
+   * All projected columns are null and the data types are come from the passed {@link RelDataType}.
+   * <pre>
+   * SELECT NULL alias0 ... NULL aliasn LIMIT 0;
```

Comment:
I added the casts but not the insert into part because it has a little bit different AST.
```
   TOK_QUERY
      TOK_INSERT
         TOK_INSERT_INTO
            TOK_TAB
               TOK_TABNAME
                  TMP_FILE
...
```
So 
```
EXPLAIN AST SELECT CAST(NULL AS type0) alias0 ... CAST(NULL AS typen) aliasn LIMIT 0;
```
has the same AST mentioned in this javadocs.

Reply:
So the AST that we are generating is equivalent to `EXPLAIN AST` query; that's interesting. Maybe worth adding this detail as well but not a must.

Reply:
I am saying that it is interesting cause I assume that executing the query:
```sql
SELECT CAST(NULL AS type0) alias0 ... CAST(NULL AS typen) aliasn LIMIT 0;
```
vs
```sql
EXPLAIN AST SELECT CAST(NULL AS type0) alias0 ... CAST(NULL AS typen) aliasn LIMIT 0;
```
has different performance implications.

Reply:
Please ignore my last two comments above. I misread your last comment. There is nothing more to do here.